tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -257,9 +266,19 @@ class RNTupleChainProcessor : public RNTupleProcessor {
    friend class RNTupleProcessor;
 
 private:
-   NTupleSize_t ConnectNTuple(const RNTupleSourceSpec &ntuple) final;
-   void ConnectFields() final;
    NTupleSize_t Advance() final;
+   void LoadEntry() final { fEntry->Read(fLocalEntryNumber); }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Connect an RNTuple for processing.
+   ///
+   /// \param[in] ntuple The RNTupleSourceSpec describing the RNTuple to connect.
+   ///
+   /// \return The number of entries in the newly-connected RNTuple.
+   ///
+   /// Creates and attaches new page source for the specified RNTuple, and connects the fields that are known by
+   /// the processor to it.
+   NTupleSize_t ConnectNTuple(const RNTupleSourceSpec &ntuple);
```

Comment:
Just as curiosity, why does this return the number of entries in the connected RNTuple?

Reply:
This is used in the `RNTupleChainProcessor` to detect whether the next ntuple is empty or not: https://github.com/root-project/root/blob/99ee076131f61a5c9f430873c9352178b25deb9a/tree/ntuple/v7/src/RNTupleProcessor.cxx#L111 (N.B., currently if this is the case it just ignores it and goes onto the next one. We might want to change this to an exception but that's otherwise unrelated to this PR)